src/helpers/object-settings.ts

```diff
@@ -41,14 +41,55 @@ export class ObjectSettings extends Array<ObjectSetting> {
         );
     }
 
-    getFieldType() {
-        return this.find(s => s.kind === 'FieldType');
-    }
+    getFieldType(args?: {
+        name: string;
+        input?: boolean;
+        output?: boolean;
+    }): ObjectSetting | undefined {
+          if (!args) return this.find(s => s.kind === 'FieldType');
+
+          const fieldType = this.find(s => s.kind === 'FieldType');
+
+          if (fieldType?.input && args.input) {
+                  return fieldType;
+          }
 
-    getPropertyType() {
-        return this.find(s => s.kind === 'PropertyType');
+          if (fieldType?.output && args.output) {
+                  return fieldType;
+          }
+
+          if (fieldType?.match?.(args.name)) {
+                  return fieldType;
+          }
+
+          // eslint-disable-next-line consistent-return
+          return undefined;
     }
 
+    getPropertyType(args?: {
```

Comment:
shouldHideField, getFieldType, getPropertyType accepts same type of argument. Maybe create type.